### DISCUSSION TITLE
[Feature] Features I would love to see in Immich. (Photo tagging, slideshow improvements, video view, date navigation, etc)

### DISCUSSION THREAD

**Discussion by mckayc** [immich-app/immich#7100] [Feature Request]
### The feature

First off, to any dev reading this. THANK YOU! I use Immich every day and absolutely LOVE it. I am super impressed with the speed of releases and all the features Immich already has. As I use it daily I keep on thinking of features I would love to see implemented. 

My expectation with this feature(s) request is not to have all the ideas implemented but I hope that one of my ideas might resonate with a developer so that the feature (or something similar) gets implemented at some point. 

I guess this is more of a feedback post from someone who uses Immich daily. If it would be better for me to do multiple separate posts for my thoughts, let me know.

**Feature Requests:**

- Ability to navigate by date - One issue I have is that I have thousands of photos and when I try to scroll to a specific date it takes me a long time. I use the scroll bar to get close but to get to the specific date I have to use the scroll wheel. What would be nice is if there was a date picker icon at the top right above the timeline. When you click on it, it opens a date picker and when you select a date and click on OK it takes you to that specific date in the timeline.

- An option to show only videos, only photos, or Motion photos. I do a lot of video editing so it would be nice if I could scroll through my timeline and only see videos so I could download what I need. I envision if you click on "Explore" under "Places" you could have three thumbnails: "Show only image files", "Show only motion photos" and "Show only video files". (Another photo service that will not be named does something similar to this with Explore>Categories).

- Tagging and Star Rating - A lot of image viewers have start ratings (1-5). It would be nice if we could do the same in Immich and the data is saved to the exif data in the photo. Similarly it would be great to be able to add tags to the exif data as well.

- Dynamic albums - If photos could use tags and stars it would be great to add dynamic albums. For example, you could create a 5 star album that gets updated dynamically as you rate photos.

- Album / Slideshow Toggle - It would be cool  if in the "Create link to share" options there was a "Share as Slideshow" so that when someone clicks on the link it automatically starts a slideshow. This would be especially nice for bloggers who want to embed a slideshow (speaking of... is there an option to share as an iFrame?).

- Description Text shown in slideshow - Each photo has the option in the Info to "Add a description". It would be nice if when you are in slideshow mode, there was an option to "Show description" that would overlay text at the bottom of each photo as it is going through the slideshow. This would be great if you are sharing a vacation slideshow and you want others to be able to read a little about it as they are watching the photos.

- Slideshow time options - It would be great if you could set how many seconds you want each photo to show for when you are in slideshow mode.

- Slideshow image navigation - It would be great if when you move your mouse cursor to the bottom of the slideshow it had a carousel of the images in the slideshow so you could navigate that way if you did not want to skip through the photos one by one.

- Age feature - One thing I like to do with my kids is to compare their photos when they are all the same age. What I imagine you could do in Immich is in "Explore" you could have a "View people at the same age" link or something like that. Once you click on the link, it gives you checkboxes of people and allows you to select an age in years and months. It will look for photos of all the people you selected and find a photos of them within a 30 day range of what you selected so that you can easily view what people looked like at the same age. Anyone without a photo of that age (within 30 days) will not show up.

- Option to toggle an avatar on top of the photo of the person who uploaded the photo - I share Immich with my wife and I have it set so both of our photos show up when we go to "Photos". Sometime I don't remember who took what photo. It would be nice to have a toggle option to show an avatar of my wife at bottom corner of the photos she uploaded. If you click on the avatar it takes me to only her photos.

Hopefully this long list of feature requests does not make it seem like I think Immich is lacking. I will continue to use Immich regardless. I just thought it would be nice to write down some of my thoughts in case any of my ideas turn out to be good ideas.

Thanks again for creating such an awesome tool!

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by jeremycase** [comment]
these are great ideas, but it would probably be helpful to split each one into its own feature request.

**Comment by bo0tzz** [comment]
I'm closing this request since it's mostly duplicates of existing ones.
